source/appModules/nvda.py

```diff
@@ -92,15 +88,28 @@ def event_gainFocus(self, obj, nextHandler):
 	# Silence invisible unknowns for stateChange as well.
 	event_stateChange = event_gainFocus
 
-	def event_foreground         (self, obj, nextHandler):
+	def event_foreground(self, obj, nextHandler):
 		if not gui.shouldConfigProfileTriggersBeSuspended():
 			config.conf.resumeProfileTriggers()
 		else:
-			NvdaSettingsCategoryPanel.handlePossibleProfileSwitch()
+			self.handlePossibleProfileSwitch()
+		nextHandler()
+
+	def event_nameChange(self, obj, nextHandler):
+		self.handlePossibleProfileSwitch()
 		nextHandler()
 
+	def isNvdaSettingsDialog(self, obj):
+		if not isinstance(obj, IAccessible):
+			return False
+		controlId = obj.windowHandle

```

Comment:
using controlId is a bit ambiguous here, since it is in fact a windowHandle and not a windowControlID

Reply:
Agreed.

Reply:
Yes, this was a mistake on my part. Good catch.